destination/destination.go

```diff
@@ -111,47 +110,34 @@ func (d *Destination) Open(ctx context.Context) error {
 	return nil
 }
 
-func (d *Destination) Write(ctx context.Context, records []sdk.Record) (int, error) {
-	if d.config.UseSingleShard {
-		partition := uuid.New().String()
-		var count int
-		for _, rec := range records {
-			_, err := d.client.PutRecord(ctx, &kinesis.PutRecordInput{
-				PartitionKey: &partition,
-				Data:         rec.Bytes(),
-				StreamARN:    &d.config.StreamARN,
-			})
-			if err != nil {
-				return count, fmt.Errorf("failed to put record into partition %s: %w", partition, err)
-			}
-
-			count++
-		}
-
-		return count, nil
-	}
-
+func (d *Destination) createPutRequestInput(records []sdk.Record) *kinesis.PutRecordsInput {
 	entries := make([]types.PutRecordsRequestEntry, 0, len(records))
 
-	// create the put records request
 	for _, rec := range records {
-		pKey := string(rec.Key.Bytes())
-		if len(pKey) > 256 {
-			pKey = pKey[:256]
+		partitionKey := d.config.PartitionKey
+		if partitionKey == "" {
+			partitionKey = string(rec.Key.Bytes())
+			if len(partitionKey) > 256 {
+				partitionKey = partitionKey[:256]
```

Comment:
I don't like assumptions 😄 Well, not ones that I can't influence as an end-user. And especially ones that could lead to unexpected behavior without even a warning.

If we are truncating the key, we should at least log a warning message, so the user is aware of it. Arguably, we could just not truncate it and let the request fail. The user can use processors to explicitly format records into the required format, this includes truncating the key.

Reply:
Hi @lovromazgon 

You're right, I've just added a warning log message when the partition key is truncated. However, I'm not so sure about having the user make a new processor just to setup the key.

It enforces a more complex setup. I'd argue back that if the default behaviour for partitionKeyTemplate is to print the record key back, it should should trim it (and we document this situation), just as a nice thing to do for the user. I don't think that the default connector configuration should enforce 2 decoupled pieces (processor + connector), looks more complex to me.

Reply:
Fair enough, as long as there's a warning so the user knows what's happening it's fine 👍 Can we document this behavior in the readme?

Reply:
Just documented it